src/redux/state.js

```diff
@@ -1,82 +1,126 @@
+        },
+        {
+          id: '6',
+          name: 'EFsquad',
+        },
+      ],
+
+      messages: [
+        {
+          id: '1',
+          message: 'qq all',
+        },
+        {
+          id: '2',
+          message: 'hello world',
+        },
+        {
+          id: '3',
+          message: 'priveti4 ot Marmeladi4a',
+        },
+      ],
+
+      newMessageBody: '',
+
+    },
   },
-  dialogsPage: {
-    dialogs: [
-      {
-        id: '1',
-        name: '105',
-      },
-      {
-        id: '2',
-        name: 'FrinteZz',
-      },
-      {
-        id: '3',
-        name: 'Gaming',
-      },
-      {
-        id: '4',
-        name: 'Pominki',
-      },
-      {
-        id: '5',
-        name: 'Side',
-      },
-      {
-        id: '6',
-        name: 'EFsquad',
-      },
-    ],
-
-    messages: [
-      {
-        id: '1',
-        message: 'qq all',
-      },
-      {
-        id: '2',
-        message: 'hello world',
-      },
-      {
-        id: '3',
-        message: 'priveti4 ot Marmeladi4a',
-      },
-    ],
+
+  getState() {
+    return this._state;
   },
-};
 
-export const addPost = () => {
-  const newPost = {
-    id: 7,
-    message: state.profilePage.newPostText,
-    likesCount: 0,
-  };
+  _callSubscriber() {
+    console.log('changed');
+  },
 
-  state.profilePage.posts.push(newPost);
-  state.profilePage.newPostText = '';
-  rerenderEntireTree(state);
-};
+  subscribe(observer) {
+    this._callSubscriber = observer;
+  },
+
+  dispatch(action) {
+    if (action.type === ADD_POST) {
+      const newPost = {
+        id: 7,
+        message: this._state.profilePage.newPostText,
+        likesCount: 0,
+      };
 
-export const updateNewPostText = (newText) => {
-  state.profilePage.newPostText = newText;
-  rerenderEntireTree(state);
+      this._state.profilePage.posts.push(newPost);
+      this._state.profilePage.newPostText = '';
+      this._callSubscriber(this._state);
+    } else if (action.type === UPDATE_NEW_POST_TEXT) {
+      this._state.profilePage.newPostText = action.newText;
+      this._callSubscriber(this._state);
+    } else if (action.type === UPDATE_NEW_MESSAGE_BODY) {
+      this._state.dialogsPage.newMessageBody = action.body;
+      this._callSubscriber(this._state);
+    } else if (action.type === SEND_MESSAGE) {
+      const body = this._state.dialogsPage.newMessageBody;
+      this._state.dialogsPage.newMessageBody = '';
+      this._state.dialogsPage.messages.push({ id: 8, message: body });
+      this._callSubscriber(this._state);
+    }
+  },
 };
 
+export const addPostActionCreator = () => ({ type: ADD_POST });
+
+export const updateNewPostTextActionCreator = (text) => ({
+  type: UPDATE_NEW_POST_TEXT,
+  newText: text,
+});
+
+export const sendMessageCreator = () => ({ type: SEND_MESSAGE });
+
+export const updateNewMessageBodyCreator = (body) => ({
+  type: UPDATE_NEW_MESSAGE_BODY,
+  body,
+});
 
-export default state;
+export default store;
```

Comment:
And given the above suggestion about the export object, try googling "export default and not default from one file"

Reply:
/fxd